### PR TITLE
CFE-1058: Add tech-preview e2e in CI config for cluster-dns-operator

### PR DESCRIPTION
**Pull request by arkadeepsen**
This PR adds CI configuration for running tech-preview e2e test in AWS for cluster-dns-operator.

### CODE REVIEW

File: ci-operator/config/openshift/cluster-dns-operator/openshift-cluster-dns-operator-master.yaml [hunk 1]
```diff
@@ -75,6 +75,13 @@ tests:
   steps:
     cluster_profile: aws
     workflow: openshift-e2e-aws-single-node
+- as: e2e-aws-ovn-techpreview
+  optional: true
```
Comment (arkadeepsen) [inline ci-operator/config/openshift/cluster-dns-operator/openshift-cluster-dns-operator-master.yaml:79]: Made the CI test optional for merging as there are higher chances of unrelated failures due to all tech-preview features getting enabled.

File: ci-operator/jobs/openshift/cluster-dns-operator/openshift-cluster-dns-operator-master-presubmits.yaml [hunk 2]
```diff
@@ -313,6 +313,85 @@ presubmits:
         secret:
           secretName: result-aggregator
     trigger: (?m)^/test( | .* )e2e-aws-ovn-single-node,?($|\s.*)
+  - agent: kubernetes
+    always_run: true
```
Comment (arkadeepsen) [inline ci-operator/jobs/openshift/cluster-dns-operator/openshift-cluster-dns-operator-master-presubmits.yaml:317]: Should we make `always_run` as false as not all changes will be related to tech-preview? We can run the test using `/test` whenever this test is needed to be run.

I had added similar test for cluster-network-operator here: https://github.com/openshift/release/pull/51201/files#diff-a1c0d571415439fe740da85b8546937cad301f1fe6adbf9c91efc365dc5f4164R1416

Comment (candita) [reply ci-operator/jobs/openshift/cluster-dns-operator/openshift-cluster-dns-operator-master-presubmits.yaml:317]: In the early stages, let's have it `always_run: true` for visibility.  Automatic CI is the de facto standard.

**Comment by arkadeepsen** [comment]
As pj-rehearse ran successfully adding the label
 /pj-rehearse ack

**Pull request merged**